arch/rv32i/src/pmp.rs

```diff
@@ -142,14 +144,11 @@ impl NAPOTRegionSpec {
     /// rejects the `pmpaddr` (tests whether any of the 10 most significant bits
     /// are non-zero).
     pub fn from_pmpaddr_csr(pmpaddr: usize) -> Option<Self> {
-        // On 64-bit platforms, the 10 most significant bits must be 0:
-        if core::mem::size_of::<usize>() == core::mem::size_of::<u64>() {
-            if (pmpaddr as u64) & !PMPADDR_RV64_MASK != 0 {
-                return None;
-            }
-        }
-
-        Some(NAPOTRegionSpec { pmpaddr })
+        // On 64-bit platforms, the 10 most significant bits must be 0.
+        // Prevent the `&-masking with zero` lint error in case of RV32.
+        // The redundant checks in this case are optimized out by the compiler on any 1-3,z opt-level.
+        #[allow(clippy::bad_bit_mask)]
+        (pmpaddr & !PMPADDR_RV64_MASK == 0).then_some(NAPOTRegionSpec { pmpaddr })
```

Comment:
I still don't like it much, but I'm fine with leaving this using `.then_some(...)`.